packages/ui/src/Badge.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FC, ReactNode } from 'react'
+import cn from 'classnames'
+import { AlertTriangle, CheckCircle, AlertCircle, Info, Icon as IconType } from 'react-feather'
+
+import { Icon } from '../src/Icon'
+
+import styles from './Badge.module.scss'
+
+export type BadgeType = 'neutral' | 'success' | 'info' | 'warning' | 'critical'
+
+export type IconDescriptor = {
+  icon: IconType
+  ariaLabel: string
+}
+
+export const BadgeIcon: { [key in BadgeType]: IconDescriptor } = {
+  neutral: {
+    icon: Info,
+    ariaLabel: 'Info circle icon',
+  },
+  success: {
+    icon: CheckCircle,
+    ariaLabel: 'Check circle icon',
+  },
+  info: {
+    icon: Info,
+    ariaLabel: 'Info circle icon',
+  },
+  warning: {
+    icon: AlertTriangle,
+    ariaLabel: 'Warning triangle icon',
+  },
+  critical: {
+    icon: AlertCircle,
+    ariaLabel: 'Info critical circle icon',
+  },
+}
+
+export type BadgeSize = 'small' | 'normal'
+
+export type BadgeProps = {
+  type: BadgeType
+  size?: BadgeSize
```

Comment:
GJ @jankoritak 👏 

I have one concern about this specific property. Do we want it? Shouldn't we leave it to the user to decide what is the font-size and inherit it? + define vertical-align to baseline?

I like how Bootstrap does it and it feels more natural https://getbootstrap.com/docs/4.5/components/badge/

We use these badges exactly like this in .cloud (part of some headings etc. and I'd like to avoid changing the property every time I change the semantics of h1-h6. I think it should work OOTB in the headings - regular text. 
WDYT?

Reply:
That's a good idea 👍 It would certainly feel more natural.

Reply:
We can do that.

However, this component currently always renders an icon with the content, does that work for your use-case?

Reply:
We'll see about how it first with the future design, however, an icon is just SVG so i could follow the same rules?

Reply:
Sure, we can make the inheritance work, I'm just curious about the icon, as Bootstrap's Badge looks a bit different.

Reply:
Addressed in: https://github.com/hazelcast/frontend-shared/pull/61/commits/59ef018563f9192c571e9c62eafde80439a8f206. Please, re-review. @jvorcak

Reply:
Guys, I am with @jankoritak on this one. Our icons come with two different sizes. I do not think we should scale them dynamically like the font size. 
On the other hand, I see the point @jvorcak is trying to make.
@jvorcak do you have places where you want a non-standard font-size for the badge? In MC they are always of the same size no matter where we place them.

Reply:
There is an option to resolve the icon size implicitly. We could inherit the font-size for the text. Then we could:
1. Run `useLayoutEffect` on the first render only.
2. Calculate the inherited `font-size` there via https://stackoverflow.com/questions/1955048/get-computed-font-size-for-dom-element-in-js
3. Based on the calculated `font-size` pass one of our existing icon sizes to the `Icon` component.
But I think we should have a valid use-case to complicate the component with this logic.

Reply:
@jankoritak @aigoncharov It'd mark this as nice-to-have, I can live with 'n' static sizes if that suites your use-case better.

The reason why I wanted this is that badges (with and without icons) are often used in headings (h1-h6) and I expect to look nice OOTB. Since an icon is SVG and the rest of the component is just text, I see no reason why we would need JS for this.

I try to see how other component libs solve the issues we have, Bootstrap handles this as I showed earlier, which I think is quite elegant. *On the other hand* Material-ui has a [chip](https://material-ui.com/components/chips/#chip) component with defined static sizes. So it really depends on how do we want to use it.

Reply:
It seems other libs sometimes even have two options, depending on whether this is semantically used as a standalone component or just an addition to a text. We don't have a lot of cases like this on .cloud, 1-3, so please feel free to pick whichever suits your need in a better way.

Reply:
@jankoritak @jvorcak I would suggest sticking to static sizes until the need of the dynamic inherited size arises. Once it does, we will add the `inherit` size prop for that use-case. WDYT?

Reply:
As for scaling of the icon, we would not need JS there, the current state (using `em`) works just fine. Although it may look a bit confusing, as we're overriding the `size` prop of the `Icon` via CSS.

I see we have a consensus now though, let me roll back to static sizes then.